python/modem-firmware-1.3+/gather_attestation_tokens.py

```diff
@@ -0,0 +1,469 @@
+                        default="attestation_tokens.csv")
+    parser.add_argument("-o", "--overwrite",
+                        help="When saving onboarding CSV, overwrite it",
+                        action='store_true', default=False)
+    parser.add_argument("--keep",
+                        help="When appending to CSV files: if UUID exists in file, keep old data not current",
+                        action='store_true', default=False)
+    parser.add_argument("--port", type=str,
+                        help="Specify which serial port to open, otherwise pick from list",
+                        default=None)
+    parser.add_argument("-A", "--all",
+                        help="List ports of all types, not just Nordic devices",
+                        action='store_true', default=False)
+    parser.add_argument("-v", "--verbose",
+                        help="bool: Make output verbose",
+                        action='store_true', default=False)
+    parser.add_argument("-P", "--plain",
+                        help="bool: Plain output (no colors)",
+                        action='store_true', default=False)
+    parser.add_argument("--xonxoff",
+                        help="Enable software flow control for serial connection",
+                        action='store_true', default=False)
+    parser.add_argument("--rtscts_off",
+                        help="Disable hardware (RTS/CTS) flow control for serial connection",
+                        action='store_true', default=False)
+    parser.add_argument("--dsrdtr",
+                        help="Enable hardware (DSR/DTR) flow control for serial connection",
+                        action='store_true', default=False)
+    parser.add_argument("--rtt",
+                        help="Use RTT instead of serial. Requires application configured for RTT console",
+                        action='store_true', default=False)
+    parser.add_argument("--jlink_sn", type=int,
+                        help="Serial number of J-Link device to use for RTT; optional",
+                        default=None)
+    parser.add_argument("--shell",
+                        help="Assume using provisioning shell",
+                        action='store_true', default=False)
+    args = parser.parse_args()
+    verbose = args.verbose
+    return args
+
+def ensure_lf(line):
+    global lf_done
+    done = lf_done
+    lf_done = True
+    return '\n' + line if not done else line
+
+def local_style(line):
+    return ensure_lf(Fore.CYAN + line
+                     + Style.RESET_ALL) if not plain else line
+
+def hivis_style(line):
+    return ensure_lf(Fore.MAGENTA + line
+                     + Style.RESET_ALL) if not plain else line
+
+def send_style(line):
+    return ensure_lf(Fore.BLUE + line
+                     + Style.RESET_ALL) if not plain else line
+
+def error_style(line):
+    return ensure_lf(Fore.RED + line + Style.RESET_ALL) if not plain else line
+
+def ask_for_port(selected_port, list_all):
```

Comment:
can you create a ticket to move these duplicate functions to a shared file for ease of maintenance?

i see we already missed updating the usb_patterns in claim_and_provision

Reply:
yes, that was apparent, just didn't take the time to do it -- I'll make a ticket

Reply:
https://nordicsemi.atlassian.net/browse/IRIS-9710

Reply:
https://nordicsemi.atlassian.net/browse/IRIS-9710